src/core/sr.ml

```diff
@@ -17,11 +17,11 @@ type subst = tvar array * term array
     constraints [cs]. The returned substitution is given by a couple of arrays
     [(xs,ts)] of the same length.  The array [xs] contains the variables to be
     substituted using the terms of [ts] at the same index. *)
-let subst_from_constrs : (term * term) list -> subst = fun cs ->
+let subst_from_constrs : unif_constrs -> subst = fun cs ->
   let rec build_sub acc cs =
     match cs with
-    | []        -> List.split acc
-    | (a,b)::cs ->
+    | []          -> List.split acc
+    | (_,a,b)::cs ->
       match Basics.get_args a with
```

Comment:
The use of get_args here is incorrect. We need to use a recursive function ctx_get_args that does the same thing as get_args + unfold the head if it is a defined variable.
In fact, we should check every where if get_args shouldn't be replaced by ctx_get_args.

Reply:
I looked rapidly, and perhaps it could be used in `unif.ml` in the `inverse` functions.

Reply:
No, I don't think so.